### PR TITLE
DOC-8272-C4-Android --N1QL Code Snippets

### PR DESCRIPTION
**Pull request by ibsoln**
https://issues.couchbase.com/browse/DOC-8272

### CODE REVIEW

File: modules/java/examples/code_snippets/Examples.java [hunk 1]
```diff
@@ -3142,30 +3142,68 @@ public void testQuerySyntaxId() throws CouchbaseLiteException {
...
 
 
-        // tag::query-syntax-pagination[]
-        try {
-          this_Db = new Database("hotels");
-        } catch (CouchbaseLiteException e) {
-          e.printStackTrace();
-        }
+      // tag::query-syntax-pagination[]
+      try {
```
Comment (bmeike) [inline modules/java/examples/code_snippets/Examples.java:3146]: In Kotlin code, you can probably just leave these try/catch block out. completely.  ... especially ones that don't do any more than print the stack trace and ignore the error

File: modules/java/examples/code_snippets/Examples.java [hunk 2]
```diff
@@ -3142,30 +3142,68 @@ public void testQuerySyntaxId() throws CouchbaseLiteException {
...
+    {
+      // For Documentation -- N1QL Query using parameters
+      // tag::query-syntax-n1ql-params[]
+      //  Declared elsewhere: Database argDB
+
+      Database thisDb = argDB;
+
+      Query thisQuery =
+          thisDb.createQuery(
+              "SELECT META().id AS thisId FROM _ WHERE type = $type"); // <.
```
Comment (bmeike) [inline modules/java/examples/code_snippets/Examples.java:3198]: Kotlin interprets a dollar sign ("$") inside a string as a variable reference.  This code will fail to compile with a message like "undefined variable 'type'".  The dollar sign needs to be escaped:

"SELECT META().id AS thisId FROM _ WHERE type = \$type")

**Review by bmeike — COMMENTED** [review]
looks fine with the two noted exceptions

File: modules/java/examples/code_snippets/Examples.java (hunk 1 above)
Comment (ibsoln) [reply modules/java/examples/code_snippets/Examples.java:3146]: Will do

File: modules/java/examples/code_snippets/Examples.java (hunk 2 above)
Comment (ibsoln) [reply modules/java/examples/code_snippets/Examples.java:3198]: This is Java @bmeike so don't need to escape it here do I? But will ensure that it is done in Kotlin module

Comment (bmeike) [reply modules/java/examples/code_snippets/Examples.java:3198]: Duh.  This, clearly, *is* Java.  Just ignore the comment.

Comment (ibsoln) [reply modules/java/examples/code_snippets/Examples.java:3198]: Dude ... if my Javotlin revolution is winning you over, you ought to ask Jim about my C+-

**Pull request merged**